addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -451,6 +451,10 @@ var SnippetEditor = publicWidget.Widget.extend({
             // body. If the editable has options, we do not want to show them.
             parent = $(parent).closest('body');
         }
+        const nextSelectedSnippet = previousSibling || nextSibling;
+        if (nextSelectedSnippet) {
+            nextSelectedSnippet.dataset.shapeAfterRemove = previousSibling ? "previous" : "next";
+        }
```

Comment:
Let's do that (updating neighbors shapes on removing a snippet) in the onRemove of the delete snippet's option